ibis/backends/druid/__init__.py

```diff
@@ -107,33 +110,74 @@ def _metadata(self, query: str) -> Iterable[tuple[str, dt.DataType]]:
                 dtype = DruidType.from_string(typ)
             yield name, dtype
 
-    def _get_temp_view_definition(
-        self, name: str, definition: sa.sql.compiler.Compiled
-    ) -> str:
-        raise NotImplementedError()
-
-    def _has_table(self, connection, table_name: str, schema) -> bool:
-        t = sa.table(
-            "TABLES", sa.column("TABLE_NAME", sa.TEXT), schema="INFORMATION_SCHEMA"
+    def get_schema(
+        self, table_name: str, schema: str | None = None, database: str | None = None
+    ) -> sch.Schema:
+        name_type_pairs = self._metadata(
+            sg.select(STAR)
+            .from_(sg.table(table_name, db=schema, catalog=database))
+            .sql(self.compiler.dialect)
         )
-        query = sa.select(
-            sa.func.sum(sa.cast(t.c.TABLE_NAME == table_name, sa.INTEGER))
-        ).compile(dialect=self.con.dialect)
-
-        return bool(connection.execute(query).scalar())
-
-    def _get_sqla_table(
-        self, name: str, autoload: bool = True, **kwargs: Any
-    ) -> sa.Table:
-        with warnings.catch_warnings():
-            warnings.filterwarnings(
-                "ignore",
-                message="|".join(  # noqa: FLY002
-                    (
-                        "Did not recognize type",
-                        "Dialect druid:rest will not make use of SQL compilation caching",
-                    )
-                ),
-                category=sa.exc.SAWarning,
+        return sch.Schema.from_tuples(name_type_pairs)
+
+    def _fetch_from_cursor(self, cursor, schema: sch.Schema) -> pd.DataFrame:
+        import pandas as pd
+
+        from ibis.formats.pandas import PandasData
+
+        try:
+            df = pd.DataFrame.from_records(
+                cursor, columns=schema.names, coerce_float=True
             )
-            return super()._get_sqla_table(name, autoload=autoload, **kwargs)
+        except Exception:
+            # clean up the cursor if we fail to create the DataFrame
+            #
+            # in the sqlite case failing to close the cursor results in
```

Comment:
sqlite case?

Reply:
yeah, copypasta. ideally we can clean this up a bit later

Reply:
though not urgent by any means